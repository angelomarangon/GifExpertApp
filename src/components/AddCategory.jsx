import React, { useState } from 'react'
import  PropTypes  from "prop-types";

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = (evento) => {

        setInputValue(event.target.value);
    };

    const onSubmit = (e)=> {
        //para evitar que cuando hacemos enter se refresque el input
        e.preventDefault();
        if(inputValue.trim().length <=1) return;

        onNewCategory(inputValue.trim());

        //de esta forma limpiamos el input cada vez que le damos enter
        setInputValue('');
    }

    return (
        <form aria-label='form' onSubmit={onSubmit}>
            <input
                type="text"
                placeholder='Buscar Gif'
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}


AddCategory.propTypes = {

    onNewCategory: PropTypes.func.isRequired,

}

