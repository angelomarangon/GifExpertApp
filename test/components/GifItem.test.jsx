import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem"


describe('Prueba en GifItem', () => {
  
    const title = 'Titulo del test';
    const url = 'https://www.asdagasfasfasdasdas.com/';

    test('Debe hacer match con el snapshot', () => {
      
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();


    })

    test('Debe de mostrar la imagen con el url y el alt indicado', () => {
      
        render(<GifItem title={title} url={url}/>);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);


        //otra forma de realizarlo

        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);


    })


    test('Debe de mostrar el titulo en el parrafo', () => {
      
        render(<GifItem title={title} url={url}/>);
        
        expect(screen.getByRole('paragraph').innerHTML).toContain(title);


    })
    

})
