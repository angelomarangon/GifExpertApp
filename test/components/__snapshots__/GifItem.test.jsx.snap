// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo del test"
      src="https://www.asdagasfasfasdasdas.com/"
    />
    <p>
      Titulo del test
    </p>
  </div>
</div>
`;
